controller/src/main/java/io/pravega/controller/task/KeyValueTable/TableMetadataTasks.java

```diff
@@ -0,0 +1,271 @@
+        this.kvtMetadataStore = kvtMetadataStore;
+        this.segmentHelper = segmentHelper;
+        this.executor = executor;
+        this.eventExecutor = eventExecutor;
+        this.hostId = hostId;
+        this.authHelper = authHelper;
+        this.requestTracker = requestTracker;
+    }
+
+    @Synchronized
+    public void initializeStreamWriters(final EventStreamClientFactory clientFactory,
+                                        final String streamName) {
+        this.requestStreamName = streamName;
+        requestEventWriterRef.set(clientFactory.createEventWriter(requestStreamName,
+                ControllerEventProcessors.CONTROLLER_EVENT_SERIALIZER,
+                EventWriterConfig.builder().build()));
+        writerInitFuture.complete(null);
+    }
+
+    /**
+     *  Create a Key-Value Table.
+     *
+     * @param scope      scope name.
+     * @param kvtName    KVTable name.
+     * @param kvtConfig  KVTable configuration.
+     * @param createTimestamp  KVTable creation timestamp.
+     * @return update status.
+     */
+    public CompletableFuture<CreateKeyValueTableStatus.Status> createKeyValueTable(String scope, String kvtName,
+                                                                                   KeyValueTableConfiguration kvtConfig,
+                                                                                   final long createTimestamp) {
+        final long requestId = requestTracker.getRequestIdFor("createKVTable", scope, kvtName);
+        return RetryHelper.withRetriesAsync(() -> Futures.exceptionallyExpecting(kvtMetadataStore.getState(scope, kvtName, true, null, executor),
+                e -> Exceptions.unwrap(e) instanceof StoreException.DataNotFoundException, KVTableState.UNKNOWN)
+                .thenCompose(state -> {
+                    if (state.equals(KVTableState.UNKNOWN) || state.equals(KVTableState.CREATING)) {
+                        return Futures.completeOn(kvtMetadataStore.checkScopeExists(scope)
+                                .thenCompose(exists -> {
+                                    if (exists) {
+                                        return Futures.exceptionallyExpecting(kvtMetadataStore.createEntryForKVTable(scope, kvtName, executor)
+                                                        .thenCompose(v -> kvtMetadataStore.getKVTable(scope, kvtName, null).getId()
+                                                        .thenCompose(uuid -> {
+                                                            CreateTableEvent event = new CreateTableEvent(scope, kvtName, kvtConfig.getPartitionCount(),
+                                                                                                        createTimestamp, requestId, UUID.fromString(uuid));
+                                                            return addIndexAndSubmitTask(event,
+                                                                    () -> CompletableFuture.completedFuture(null))
```

Comment:
the addIndexAndSubmitTask method gives atomic guarantee of posting event and updating a metadata.. 
you are not updating any metadata inside this. you are posting an event though.. so you can simply post the event in that case directly. you dont need addIndexAndSubmitTask method. basically then what it means is even if we create an entry into the KV tables, if we fail to post the event, the create is never processed. this is fine. the caller will get an error resulting from failure to post event anyway (or connection failure if controller crashed). 

Alternatively, if you want to use the atomic guarantee of addIndexAndSubmitTask method you can also do the following:
 
```
UUID = new UUID(); // generate the id here instead of inside the store. and pass it to the store and include it in the event. 
addIndexAndSubmitTask(event, kvtMetadataStore.createEntryForKVTable(scope, kvtName, uuid, executor))
```
notice that i passed the ID to the kvmetadata store .. and if an id is already generated and exists in the table, this will fail with write conflict exception.. 
you can decide on the handling for the conflict - basically if another id is written, its event must have been posted too, and it will eventually complete.. so you can simply wait for the create to complete and respond with `success` or `existing` depending on what got created. 

My personal preference will be the latter approach.

Reply:
Sure. Made changes as per latter approach.

Reply:
i still see 
```
                                                            return addIndexAndSubmitTask(event,
                                                                    () -> CompletableFuture.completedFuture(null)) 
```